saleor/graphql/discount/mutations/sale/sale_base_catalogue.py

```diff
@@ -19,3 +26,30 @@ class Arguments:
 
     class Meta:
         abstract = True
+
+    @classmethod
+    def post_save_actions(
+        cls, info: ResolveInfo, promotion, previous_catalogue, new_catalogue
+    ):
+        if previous_catalogue != new_catalogue:
+            manager = get_plugin_manager_promise(info.context).get()
+            cls.call_event(
+                manager.sale_updated,
+                promotion,
+                previous_catalogue,
+                new_catalogue,
+            )
+
+        previous_predicate = convert_catalogue_info_into_predicate(previous_catalogue)
+        new_predicate = convert_catalogue_info_into_predicate(new_catalogue)
+        previous_product_ids = get_product_ids_for_predicate(previous_predicate)
+        new_product_ids = get_product_ids_for_predicate(new_predicate)
+
+        if previous_product_ids != new_product_ids:
+            if len(new_product_ids) > len(previous_product_ids):
+                product_ids = new_product_ids - previous_product_ids
+            else:
+                product_ids = previous_product_ids - new_product_ids
+            update_products_discounted_prices_for_promotion_task.delay(
+                list(product_ids)
+            )
```

Comment:
We should recalculate prices for both new and removed product ids

Reply:
Hmm, logic behind is:

-  `if len(new_product_ids) > len(previous_product_ids) is True` it means this is `saleAddCatalogue` mutation, so we only need to recalculate products we just added. Products, which were previously in catalogue don't need to be recalculated, thus `product_ids = new_product_ids - previous_product_ids`
- `if len(new_product_ids) > len(previous_product_ids) is False` it means this is `saleRemoveCatalogue` mutation, so we only need to recalculate products we just removed. There is no need to recalculate products, which are still in catalogue, thus `product_ids = previous_product_ids - new_product_ids`.

Does it make sense?

Reply:
Ok, right now it has sense, but please add a comments to explain that :)

Reply:
added explicit flag `is_add_mutation`